### DISCUSSION TITLE
Modifying and compiling source code

### DISCUSSION THREAD

**Discussion by LNTU-YS** [groops-devs/groops#221] [Q&A] [answered]
Hello,

I am a beginner in programming on QT. Now I have encountered some problems and I don't know what to do. I hope to get your help.

For example, if I want to modify the code for the process of 'orbit2Sp3Format', I think I should make the modifications in 'groops\source\programs\conversion\orbit2Sp3Format.cpp'. But after I make changes to this file, QT will prompt me with 'Warning: This file is outside the project directory'. I have also attempted to write the “. h” and “. cpp” files in the "groops\source" folder to the corresponding location in "groovsGui. pro", but there have been many errors and issues. When I don't make any modifications, I can successfully build projects on QT.

So what should I do to successfully modify the program on QT.

Thank you in advance for your feedback and help.
With Kind Regards
YUAN

**Comment by tmayerguerr** [comment, marked as answer]
Hi Yuan,

QT is not needed to compile the GROOPS source. The following steps must be completed:

1.  (Maybe add the additional source .cpp files in `source/sourcesCXX.txt`)

2. Compile the code as described in the [installation page](https://github.com/groops-devs/groops/blob/main/INSTALL.md) (here for Windows): Open the MSYS2 MinGW 64-bit terminal through 
```
Start Menu > MSYS2 64-bit > MSYS2 MinGW 64-bit.
```

3. Create the build directory and compile GROOPS:
```
mkdir /c/groops/source/build && cd /c/groops/source/build
cmake.exe .. -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="C:\groops"
mingw32-make.exe -j4
mingw32-make.exe install 
```

4. Create the schema file used by the GUI and to update the documentation.
```
cd /c/groops/
bin/groops.exe --xsd groops.xsd
bin/groops.exe --doc docs/
```

Best regards
Torsten

**Reply by LNTU-YS** [reply]
Hallo Torsten

Thank you for the feedback and the suggestion. I've tried and got the results I wanted. 

With Kind Regards
YUAN
